doc/hardware/porting/board_porting.rst

```diff
@@ -310,27 +438,43 @@ application for it.
 Setting Kconfig configuration values is documented in detail in
 :ref:`setting_configuration_values`.
 
-There are three mandatory Kconfig files in the board directory for a board
-named ``plank``:
+There are one mandatory Kconfig file in the board directory, and several
+optional files for a board named ``plank``:
 
 .. code-block:: none
 
-   boards/<ARCH>/plank
-   ├── Kconfig.board
+   boards/<vendor>/plank
+   ├── Kconfig
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
-   └── plank_defconfig
+   ├── plank_defconfig
+   └── plank_<identifier>_defconfig
+
+:file:`Kconfig.plank`
+  A shared Kconfig file which can be sourced both in Zephyr Kconfig and sysbuild
+  Kconfig trees.
+
+  This file select the SoC in the Kconfig tree and potential other SoC related
+  Kconfig settings. This file must not select anythink outside the re-usable
+  Kconfig board and SoC trees.
+
+  A :file:`Kconfig.plank` may look like this:
+
+  .. code-block:: kconfig
+
+     config BOARD_PLANK
+     	select SOC_SOCA
 
-:file:`Kconfig.board`
-  Included by :zephyr_file:`boards/Kconfig` to include your board
-  in the list of options.
+:file:`Kconfig`
+  Included by :zephyr_file:`boards/Kconfig`.
 
-  This should at least contain a definition for a ``BOARD_PLANK`` option,
-  which looks something like this:
+  This file can provide board specific Kconfig settings which are specific to
+  the current board.
 
+  Not all boards are having a :file:`Kconfig` file.
   .. code-block:: kconfig
 
      config BOARD_PLANK
-     	bool "Plank board"
      	depends on SOC_SERIES_YOUR_SOC_SERIES_HERE
```

Comment:
Should ` depends on SOC_SERIES_YOUR_SOC_SERIES_HERE` be kept ?

Reply:
No, should not be needed.

Removed to avoid confusion.